### DISCUSSION TITLE
Prefered cloud storage provider?

### DISCUSSION THREAD

**Discussion by AliFlux** [protomaps/PMTiles#109] [Q&A]
Hi.

In the documentation it says that: `R2 is the recommended storage platform for PMTiles`. However on the pmtiles website, digitalocean spaces are being used. Is there any specific reason why the cloud provider was switched?

I also noticed that when cloudflare R2 is being used with caching turned on (i.e. not using r2.dev URLs), then large datasets just don't load. The server returns the entire file instead of the ranged request.

**Comment by bdon** [comment]
@AliFlux R2 is recommended because its pricing model is only request-based and does not count bandwidth, which is more economical for most use cases.

DigitalOcean is used in some places on the website for testing reasons. It does work pretty well with "CDN" endpoint DigitalOcean spaces provides but this does not give the same benefits as the other integrations at http://protomaps.com/docs/cdn

Cloudflare's CDN intentionally does not work with range requests. You will need to create a Workers integration as documented in the link above to do this.
